dockstore-integration-testing/src/test/java/io/dockstore/client/cli/OrganisationIT.java

```diff
@@ -0,0 +1,308 @@
+
+        // Other user should not
+        try {
+            organisation = organisationsApiUser1.getOrganisationById(registeredOrganisation.getId());
+        } catch (ApiException ex) {
+        } finally {
+            organisation = null;
+        }
+        assertTrue("Organisation should NOT be returned.", organisation == null);
+
+        // Admin approve it
+        organisationsApiAdmin.approveOrganisation(registeredOrganisation.getId());
+
+        // Should now appear in approved list
+        organisationList = organisationsApiUser2.getApprovedOrganisations();
+        assertEquals("Should have one approved organisations." , organisationList.size(), 1);
+
+        // User should be able to get by id
+        organisation = organisationsApiUser2.getOrganisationById(registeredOrganisation.getId());
+        assertTrue("Organisation should be returned.", organisation != null);
+
+        // Other user should also be able to
+        organisation = organisationsApiUser1.getOrganisationById(registeredOrganisation.getId());
+        assertTrue("Organisation should be returned.", organisation != null);
+
+        // Admin should also be able to
+        organisation = organisationsApiAdmin.getOrganisationById(registeredOrganisation.getId());
+        assertTrue("Organisation should be returned.", organisation != null);
+
+        // Curator should be able to see
+        organisation = organisationsApiCurator.getOrganisationById(registeredOrganisation.getId());
+        assertTrue("Organisation should be returned.", organisation != null);
+
+        // Update the organisation
+        String link = "www.anothersite.com";
+        newOrganisation.setLink(link);
+        organisation = organisationsApiUser2.updateOrganisation(newOrganisation, organisation.getId());
+
+        // Organisation should have new information
+        organisation = organisationsApiUser2.getOrganisationById(registeredOrganisation.getId());
+        assertEquals("Organisation should be returned and have an updated link.", link, organisation.getLink());
+    }
+
+    /**
+     * Tests that you cannot create an organisation if another organisation already exists with the same name
+     */
+    @Test
+    public void testCreateDuplicateOrganisation() {
+        final ApiClient webClient = getWebClient(USER_2_USERNAME);
+        OrganisationsApi organisationsApi = new OrganisationsApi(webClient);
+        createOrg(organisationsApi);
+
+        boolean throwsError = false;
+        try {
+            createOrg(organisationsApi);
+        } catch (ApiException ex) {
+            throwsError = true;
+        }
+
+        if (!throwsError) {
+            fail("Was able to register an organisation with a duplicate toolname.");
+        }
+    }
+
+    /**
+     * Tests that an organisation maintainer can request a user to join.
+     * The user can then join the organisation as a member.
+     * They cannot edit anything.
+     * Change role to maintainer.
```

Comment:
Can't remember what we decided on in the specification. 
But it might be simpler if members can edit collections. 
Maintainers can do that and modify/invite organization members.

Reply:
Oh sorry in this case I meant they cannot edit metadata for the org. Unless you think that doesn't make sense.